src/openapi.ts

```diff
@@ -0,0 +1,653 @@
+        const messageName = `${operationId}Response${statusCode}`;
+        messages[messageName] = {
+          name: messageName,
+          title: `${method.toUpperCase()} response ${statusCode}`,
+          contentType: contentType,
+          payload: mediaType.schema,
+          summary: response.description,
+          headers: response.headers ? convertHeadersToSchema(response.headers) : undefined,
+        };
+      });
+    } else {
+      const messageName = `${operationId}Response${statusCode}`;
+      messages[messageName] = {
+        name: messageName,
+        title: `${method.toUpperCase()} response ${statusCode}`,
+        summary: response.description,
+      };
+    }
+  });
+
+  return messages;
+}
+
+/**
+ * Converts OpenAPI Components Object to AsyncAPI Components Object.
+ * @param {OpenAPIDocument} openapi - The complete OpenAPI document.
+ * @returns {AsyncAPIDocument['components']} The converted AsyncAPI Components Object.
+ */
+function convertComponents(openapi: OpenAPIDocument): AsyncAPIDocument['components'] {
+  const asyncComponents: AsyncAPIDocument['components'] = {};
+
+  if (openapi.components) {
+    if (openapi.components.schemas) {
+      asyncComponents.schemas = convertSchemas(openapi.components.schemas);
+    }
+
+    if (openapi.components.securitySchemes) {
+      asyncComponents.securitySchemes = convertSecuritySchemes(openapi.components.securitySchemes);
+    }
+
+    if (openapi.components.parameters) {
+      asyncComponents.parameters = {};
+      for (const [name, param] of Object.entries(openapi.components.parameters)) {
+        if (!isRefObject(param)) {
+          asyncComponents.parameters[name] = convertParameter(param);
+        } else {
+          asyncComponents.parameters[name] = param; 
+        }
+      }
+    }
+
+    if (openapi.components.responses) {
+      asyncComponents.messages = convertComponentResponsesToMessages(openapi.components.responses);
+    }
+
+    if (openapi.components.requestBodies) {
+      asyncComponents.messageTraits = convertRequestBodiesToMessageTraits(openapi.components.requestBodies);
+    }
+
+    if (openapi.components.headers) {
+      asyncComponents.messageTraits = {
+        ...(asyncComponents.messageTraits || {}),
+        ...convertHeadersToMessageTraits(openapi.components.headers)
+      };
+    }
+
+    if (openapi.components.examples) {
+      asyncComponents.examples = openapi.components.examples;  
+    }
+  }
+
+  return removeEmptyObjects(asyncComponents);
+}
+
+/**
+ * converts openAPI schema object to multiformat/schema object
+ * @param schema openAPI schema object
+ * @returns multiformat/schema object
+ */
+function convertSchema(schema: any): any {
+  if (isRefObject(schema)) {
+    // Check if it's an external reference
+    if (schema.$ref.startsWith('./') || schema.$ref.startsWith('http')) {
+      return schema; 
+    }
+    return schema;
+  }
```

Comment:
This is not entirely correct we still need to support External schema objects case, as we still need to convert the schema to multi schema object even if external. We just dont touch the external file and do any changes there. That is the restriction.


Make sure to add a test case for this.

Reply:
```ts
if (schema.$ref.startsWith('./') || schema.$ref.startsWith('http')) {
      // Convert external references to multi-format schema objects
      return {
        schemaFormat: 'application/vnd.oai.openapi;version=3.0.0',
        schema: schema
      };
    }
    return schema;
```
pushed this